src/isa/riscv64/difftest/dut.c

```diff
@@ -29,7 +29,10 @@
 
 bool isa_difftest_checkregs(CPU_state *ref_r, vaddr_t pc) {
   csr_prepare();
-  if(cpu.mip != ref_r->mip) ref_r->mip = cpu.mip; // ignore difftest for mip
+#ifdef CONFIG_DIFFTEST_REF_SPIKE
+  cpu.mip &= 0xffffff4f; // ignore difftest for mip
```

Comment:
Should we mask the bits off for `ref_r->mip` instead of zeroing these bits for cpu?

Such as `ref_r->mip &= the_mask`? Unless the cpu states will be changed by difftest. Is this expected?

Reply:
I still doubt whether we should change the `cpu.mip` instead of copying partial of it. But if you think this's ok, I'm now approving this